### PR TITLE
Add tests for read dataset branch from comment body

### PR DESCRIPTION
**Pull request by koernerfelicia**
Add tests retroactively for [this PR](https://github.com/RasaHQ/rasa/pull/10786)

### CODE REVIEW

**Comment by koernerfelicia** [comment]
This one needs to be merged after yours @markus-hinsche so we can have gomplate for GHA tests

File: .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import pathlib
+import subprocess
+import pytest
+
+TEMPLATE_FPATH = ".github/templates/model_regression_test_read_dataset_branch.tmpl"
+REPO_DIR = pathlib.Path("").absolute()
+TEST_DATA_DIR = str(pathlib.Path(__file__).parent / 'test_data')
+DEFAULT_DATASET_BRANCH = "main"
```
Comment (markus-hinsche) [inline .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py:8]: This variable `DEFAULT_DATASET_BRANCH` seems to be unused. Should it be in `parametrize`?

File: .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+@pytest.mark.parametrize(["comment_body_file", "expected_dataset_branch"],
```
Comment (markus-hinsche) [inline .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py:11]: I noticed that many of our unit tests have this format

```suggestion
@pytest.mark.parametrize("comment_body_file,expected_dataset_branch",
```

File: .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                         [
+                            ("comment_body.json", "test_dataset_branch"),
+                            ("comment_body_no_dataset_branch.json", "main")
+                         ])
+def test_read_dataset_branch(comment_body_file, expected_dataset_branch):
+    CMD = ("gomplate "
```
Comment (markus-hinsche) [inline .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py:17]: I wonder if we should lowercase `CMD` to `cmd` like with other local variables. I probably caused this by making the same mistake in https://github.com/RasaHQ/rasa/pull/10804/files

**Review by markus-hinsche — APPROVED** [review]
LGTM. Just some tiny comments about code style. 

I like how gomplate can read the JSON either from a URL or a local file. That eases testing.

File: .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py (hunk 1 above)
Comment (koernerfelicia) [reply .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py:8]: 🤦‍♀️ yep, you're very right

File: .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py (hunk 3 above)
Comment (koernerfelicia) [inline .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py:17]:
```suggestion
    cmd = ("gomplate "
```

File: .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+TEMPLATE_FPATH = ".github/templates/model_regression_test_read_dataset_branch.tmpl"
+REPO_DIR = pathlib.Path("").absolute()
+TEST_DATA_DIR = str(pathlib.Path(__file__).parent / 'test_data')
+DEFAULT_DATASET_BRANCH = "main"
+
+
+@pytest.mark.parametrize(["comment_body_file", "expected_dataset_branch"],
+                         [
+                            ("comment_body.json", "test_dataset_branch"),
+                            ("comment_body_no_dataset_branch.json", "main")
```
Comment (koernerfelicia) [inline .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py:14]:
```suggestion
                            ("comment_body_no_dataset_branch.json", DEFAULT_DATASET_BRANCH)
```

File: .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+@pytest.mark.parametrize("comment_body_file,expected_dataset_branch",
+                         [
+                            ("comment_body.json", "test_dataset_branch"),
+                            ("comment_body_no_dataset_branch.json", DEFAULT_DATASET_BRANCH)
+                         ])
+def test_read_dataset_branch(comment_body_file, expected_dataset_branch):
+    cmd = ("gomplate "
+           f"-d github={TEST_DATA_DIR}/{comment_body_file} "
+           f"-f {TEMPLATE_FPATH}")
+    output = subprocess.check_output(CMD.split(' '), cwd=REPO_DIR)
```
Comment (koernerfelicia) [inline .github/tests/test_model_regression_test_read_dataset_branch_tmpl.py:20]:
```suggestion
    output = subprocess.check_output(cmd.split(' '), cwd=REPO_DIR)
```

**Pull request merged**